### PR TITLE
Make airflow upgrade_check a command from a separate dist

### PR DESCRIPTION
**Pull request by ashb**
By doing it this way we can add new features to upgrade_check, such as
the adding a flag to make changes automatically, without having to
release a new version of Airflow.

I will next work on packaging up the existing checks in to a packageable format.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

File: airflow/bin/cli.py [hunk 1]
```diff
@@ -2888,9 +2879,10 @@ class CLIFactory(object):
         },
         {
             'name': 'upgrade_check',
-            'help': 'Check if you can upgrade to the new version.',
+            'help': 'Check if you can safely upgrade to the new version.',
             'func': upgrade_check,
-            'args': ('save', ),
+            'from_module': 'airflow.upgrade.achecker',
```
Comment (kaxil) [inline airflow/bin/cli.py]:
```suggestion
            'from_module': 'airflow.upgrade.checker',
```

File: airflow/bin/cli.py [hunk 2]
```diff
@@ -2154,16 +2152,9 @@ def info(args):
...
-        if not filename.lower().endswith(".json"):
-            print("Only JSON files are supported", file=sys.stderr)
-        formatter = JSONFormatter(args.save)
-    else:
-        formatter = ConsoleFormatter()
-    all_problems = check_upgrade(formatter)
-    if all_problems:
-        sys.exit(1)
+    sys.exit("""
+Please install apache-airflow-upgrade-check distribution from PyPi to perform upgrade checks
```
Comment (kaxil) [inline airflow/bin/cli.py]:
```suggestion
Please install apache-airflow-upgrade-check distribution from PyPI to perform upgrade checks
```
nit

File: airflow/bin/cli.py (hunk 1 above)
Comment (ashb) [reply airflow/bin/cli.py]: Whops. Testing code left over :grin:

**Review by kaxil — APPROVED** [review]

**Comment by ashb** [comment]
@potiuk Test failed with "Local airflow CI image not available" which is not an error I've seen before.

Possible we've got a bug in CI harness on v1-10-stable PRs now. Do you have time to look, or if not any ideas where I should start looking at?

**Comment by potiuk** [comment]
looking!

**Comment by potiuk** [comment]
Nope that 's not it. This message is perfectly fine there. It's a real flake8 error but I must have lost a fix not showing problems on pre-commits failing. I will make a PR shortly but in the meantime just run flake8 locally.

**Comment by potiuk** [comment]
Nope. I have not lost the messages - they are shown  all right.

I believe this is caused by https://github.com/apache/airflow/pull/12397/files#diff-2dfb149ecb4c1fe07a9c14f488a64326a8255a369d85fee621c4ecbd092a21acR2279  -> for some reason upgrade_check() is called when flake is executed likely and it causes system.exit()

**Comment by potiuk** [comment]
I also made https://github.com/apache/airflow/pull/12402 chage to hide those messages by default. They are not useful for the user.

**Comment by ashb** [comment]
> Nope. I have not lost the messages - they are shown all right.
> 
> I believe this is caused by https://github.com/apache/airflow/pull/12397/files#diff-2dfb149ecb4c1fe07a9c14f488a64326a8255a369d85fee621c4ecbd092a21acR2279 -> for some reason upgrade_check() is called when flake is executed likely and it causes system.exit()

:scream: Flake is running code? THat's horrifying. I'll take a look.

**Comment by potiuk** [comment]
or maybe simplu failing hard on `sys.exit("STRING")` !!!!

**Comment by potiuk** [comment]
Hmm. will take a look and try to reproduce

**Comment by ashb** [comment]
I'm just rebuilding the image locally to elimiate that.

**Comment by ashb** [comment]
(It's at times like this I wish pre-commit would have an option to output more debugging!

**Comment by ashb** [comment]
Hmmmm, now I've rebuilt it, pre-commit locally is showing me the error.... CURIOUS

**Comment by potiuk** [comment]
Yep. Same for me. strange :(

**Pull request merged**